keep/common/management/commands/update_https.py

```diff
@@ -0,0 +1,214 @@
+        self.summary_log.writerow(('Environments', 'PIDMAN', settings.PIDMAN_DOMAIN))
+        self.summary_log.writerow(('Time', 'Status', 'noid', \
+            'Pidman label', 'Pidman target_uri', \
+            'target_uri update', 'Exception Details'))
+
+        # collect pids in Keep Domain
+        results = self.pidman.search_pids(domain_uri=settings.PIDMAN_DOMAIN)
+        results_count = results["results_count"]
+
+        self.update_progress(results, results_count)
+
+    def update_progress(self, results, results_count):
+        """Report the objects in Pidman and summarize in a CSV.
+
+            :param results: results from pidman
+            :param results_count: total count of objects founds within a collection
+        """
+
+        # update progress on the screen
+        sys.stdout.write("%i objects in total.\n" % results_count)
+        sys.stdout.flush()
+
+        # initialize a progress bar following the Readux example
+        pbar = ProgressBar(widgets=[Percentage(),
+            ' (', Counter(), ')',
+            Bar(),
+            ETA()],
+            maxval=results_count).start()
+
+        max_results_per_page = results["max_results_per_page"]
+        pages = int(math.ceil(results_count / float(max_results_per_page))) + 1
+        current_count = 0
+        change_count = 0
+        nochange_count = 0
+
+        # iterate through all results fetched from pidman
+        for page in range(1, pages):
+            page_results = self.pidman.search_pids(domain_uri=settings.PIDMAN_DOMAIN, page=page)
+            for page_result in page_results["results"]:
+                status_label = ""
+                updated_target_uri = ""
+                pidman_target_uri = ""
+                object_label = ""
+                exception_string = ""
+
+                try:
+                    pidman_target_uri = page_result["targets"][0]["target_uri"]
+                    object_label = page_result["name"]
+                    target_uri = urlparse(pidman_target_uri)
+                    if target_uri.scheme != "https":
+                        change_count += 1
+                        https_uri = "https://" + target_uri.netloc + target_uri.path
+                        if not self.is_dry_run:
+                            response = self.pidman.update_target(type="ark", noid=page_result["pid"], target_uri=https_uri)
+                            updated_target_uri = response["target_uri"]
+                            status_label = "https-changed"
+                        else:
+                            status_label = "dry-run"
+                            updated_target_uri = https_uri
+                    else:
+                        nochange_count += 1
+                        status_label = "no-change-needed"
+
+                except Exception as e:
+                    status_label = "error"
+                    exception_string += "Pidman object %s error message: %s" % (page_result["pid"], str(e))
```

Comment:
here is another opportunity to have it skip the current one and go to the next object by inserting "break"  if that is the behavior you want.